File: solutions/julia/nth-prime/1/nth-prime.jl
function prime(num)
    if num<1
        error("No such prime!")
    else
        primes = Int[2]
        candidate = 3
        while length(primes)<=num 
            is_prime = true
            for p in primes
                if candidate%p==0 
                    is_prime = false
                    break
                end
            end
            is_prime && push!(primes, candidate)
            candidate += 2
        end
        primes[num]
    end
end

File: solutions/r/nth-prime/1/nth-prime.R
prime <- function(n) {
  if(n<1){
    stop("There is no zeroth prime")
  }
  primes <-rep(NA,times=n)
  primes[1] <- 2
  cand <-3 
  idx = 1
  while(idx<n){
    is_prime <- TRUE
    for(p in primes[1:idx]){
      if(cand %% p == 0){
        is_prime <- FALSE
        break
      }
    }
    if(is_prime){
      idx <- idx+1
      primes[idx]<-cand
     }
    cand <- cand+2
  }
  #print(primes)
  primes[n]
}
